### ISSUE TITLE
List items, knowls, MathJax

### ISSUE THREAD

**Issue by rbeezer** [rbeezer/mathbook#143]
With more than one knowl in a single list item, all but the first one opened do not process the MathJax portions.  With a page reload, you can try opening different ones first (successfully).

List at the bottom of Chapter 17 (Cross-Referencing) of the sample article demonstrates this (link is not permanent):
http://beezers.org/knowlized/derivatives.html

**Comment by rbeezer** [comment]
Alex Jordan also says:  Another bug: When the knowl for a reference to an `<men>` element is clicked to open, for some reason MathJax fails to process the math. What I see is the same kind of thing I'd see if MathJax encountered bad LaTeX syntax (code surrounded by a box). But the syntax is fine, and MathJax handles it fine in the original instance of the `<men>`.

**Comment by rbeezer** [comment]
Just tried a small expreiment with knowl `mdn-1.html` which shows up for Equation 4.2 in the list mentioned above.

Stripped out the in-contenxt link at the bottom.   (My XML parser did not like that the knowl had two top-level elements in an unrelatred experiment.)  Then stripped out the `article` element.  So all that was left was a LaTeX `align` environment.

No apparent change in behavior in either case.

**Comment by antfp** [comment]
Hi,
Discovered MathBook XML only last week, and started to experiment with it. My goal is to design a math book in the Portuguese language that renders well in html, pdf and epub. I've found that MathBook XML is a very promising project and I hope that I’ll be able to translate to xml  all the material I already wrote in html. 

I also noticed this bug, and if we remove the \label{....} for equations in knowls (and don't forget to empty the cache on the browser) the problem goes away.

Hope this helps.

**Comment by rbeezer** [comment]
Dear antfp,

Thanks very much for your interest in MathBook XML and for the sleuthing.  I was off-line all day yesterday, so just now getting caught up.

Excellent!  This one had me perplexed, and removing labels does seem to cure the problem.  I have been using the same template for math/MathJax in LaTeX and in HTML, but it now seems I need to handle the two situations differently.  So this is a very big help.

Two notes:

1.  If you have not seen, there is support for Brazilian Portugese, `pt-BR`.  Read the comments in  `xsl/mathbook-localization.xsl` if you have not found it already.

2. EPUB conversion should be easy, since it is basically HTML.  But the technical details (and reader support) is very tricky.  I know what to do, but it'll be a project.  I may use some support I have starting in January to seriously attack this.

Good luck with your writing project.  If you make something publicly available, I'd love to link to it from the project web site's "Gallery".  Feel free to use the Google Group for any problems, and especially for language and font issues.

Again - thanks for your help with this bug!

Rob

**Comment by antfp** [comment]
Dear R. Beezer,

I'm glad I could help and hope you'll find a solution soon. 

With respect to your 2 notes:

1) I'm writing in Portuguese from Portugal, so I created the localization for pt-PT and added the translations to the mathbook-localization.xsl in my project. I can share them if you wish. Just tell me how or where should I post the suggestions. By the way, maybe it's a good idea to split that file by language as new languages arrive.

2)  I will think about the EPUB only after  I have finish  the html/latex book. 

Best regards
A. Pereira

**Comment by rbeezer** [comment]
This should now be fixed at: b48cafb8

Simple fix in concept, but took some work to implement.  Thanks again, A. Pereira, I was afraid this one was going to be a battle.

**Comment by rbeezer** [comment]
On 07/24/2015 03:28 PM, antfp wrote:
> 1) I'm writing in Portuguese from Portugal, so I created the localization for
> pt-PT and added the translations to the mathbook-localization.xsl in my project.
> I can share them if you wish. Just tell me how or where should I post the
> suggestions. By the way, maybe it's a good idea to split that file by language
> as new languages arrive.

That would be excellent.  Attaching a simple text file here with the relevant 
bits (on a new issue) would be just fine.  Send me enough info to identify you 
as I have for other contributors.  If it is easier, a pull request, or email 
git-patch would be fine, too.  I can make you the author in the log no matter what.

Please use spaces for indentation, not tabs.  And yes, I expect French and 
German any day now and I should get more organized file-by-file.  Right now it 
is easier as I add new terms (like maybe "Exercise Block" for"exercisegroup" now).

> 2) I will think about the EPUB only after I have finish the html/latex book.

Depending on timing, you may not have to even think about it.  There will be a 
conversion and it will "just work."  ;-)

Rob
